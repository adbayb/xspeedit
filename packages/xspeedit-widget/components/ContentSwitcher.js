import React from "react";
import PropTypes from "prop-types";
import cn from "classnames";

const ContentSwitcher = ({ children, isActive }) => {
	return (
		<div className={cn({ active: isActive })}>
			{children}
			<style jsx>{`
				div {
					display: flex;
					flex-direction: column;
					position: absolute;
					top: 0;
					bottom: 0;
					left: 0;
					right: 0;
					background-color: #2d2e31;
					transition: all 500ms;
					background: linear-gradient(to top, #232526, #414345);
					overflow-y: auto;
				}
				.active {
					transform: translate3d(0, 0, 0);
					opacity: 1;
					visibility: visible;
				}
				:not(.active) {
					transform: translate3d(0, 50px, 0);
					opacity: 0;
					visibility: hidden;
				}
			`}</style>
		</div>
	);
};

ContentSwitcher.propTypes = {
	isActive: PropTypes.bool.isRequired,
	children: PropTypes.node
};

export default ContentSwitcher;
