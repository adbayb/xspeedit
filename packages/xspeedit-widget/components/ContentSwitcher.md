**Preview:**

```js
class Example extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			view: true
		};
		this.handleClick = this.handleClick.bind(this);
	}

	handleClick() {
		this.setState(prevState => ({
			view: !prevState.view
		}));
	}

	render() {
		const { view } = this.state;

		return (
			<div style={{ display: "flex", position: "relative", backgroundColor: "black", height: 200 }}>
				<ContentSwitcher isActive={view}>
					<div style={{ position: "relative", backgroundColor: "lightblue", flex: 1 }} />
				</ContentSwitcher>
				<ContentSwitcher isActive={!view}>
					<div style={{ position: "relative", backgroundColor: "red", flex: 1 }} />
				</ContentSwitcher>
				<button style={{ position: "absolute" }} onClick={this.handleClick}>
					Switch
				</button>
			</div>
		);
	}
}

<Example />;
```

**Example:**

```jsx static
import React from "react";

const Example = () => {
	return (
		<Fragment>
			<ContentSwitcher isActive={true}>
				<div>Content A</div>
			</ContentSwitcher>
			<ContentSwitcher isActive={false}>
				<div>Content B</div>
			</ContentSwitcher>
		</Fragment>
	);
};
```
