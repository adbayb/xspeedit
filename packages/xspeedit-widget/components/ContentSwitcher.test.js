import React from "react";
import ContentSwitcher from "./ContentSwitcher.js";

describe("ContentSwitcher", () => {
	it("should render correctly when it's not active", () => {
		// GIVEN: notre composant:
		const tree = shallow(<ContentSwitcher> Content </ContentSwitcher>);
		// THEN: il doit s'afficher correctement en matchant son snapshot:
		expect(tree).toMatchSnapshot();
	});

	it("should render correctly when it's active", () => {
		// GIVEN: notre composant:
		const tree = shallow(<ContentSwitcher isActive> Content </ContentSwitcher>);
		// THEN: il doit s'afficher correctement en matchant son snapshot:
		expect(tree).toMatchSnapshot();
	});
});
