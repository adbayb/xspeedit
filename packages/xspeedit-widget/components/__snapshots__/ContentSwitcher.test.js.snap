// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentSwitcher should render correctly when it's active 1`] = `
<div
  className="jsx-2628240846 active"
>
   Content 
  <JSXStyle
    css="div.jsx-2628240846{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;position:absolute;top:0;bottom:0;left:0;right:0;background-color:#2d2e31;-webkit-transition:all 500ms;transition:all 500ms;background:linear-gradient(to top,#232526,#414345);overflow-y:auto;}.active.jsx-2628240846{-webkit-transform:translate3d(0,0,0);-ms-transform:translate3d(0,0,0);transform:translate3d(0,0,0);opacity:1;visibility:visible;}.jsx-2628240846:not(.active){-webkit-transform:translate3d(0,50px,0);-ms-transform:translate3d(0,50px,0);transform:translate3d(0,50px,0);opacity:0;visibility:hidden;}"
    styleId="2628240846"
  />
</div>
`;

exports[`ContentSwitcher should render correctly when it's not active 1`] = `
<div
  className="jsx-2628240846 "
>
   Content 
  <JSXStyle
    css="div.jsx-2628240846{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;position:absolute;top:0;bottom:0;left:0;right:0;background-color:#2d2e31;-webkit-transition:all 500ms;transition:all 500ms;background:linear-gradient(to top,#232526,#414345);overflow-y:auto;}.active.jsx-2628240846{-webkit-transform:translate3d(0,0,0);-ms-transform:translate3d(0,0,0);transform:translate3d(0,0,0);opacity:1;visibility:visible;}.jsx-2628240846:not(.active){-webkit-transform:translate3d(0,50px,0);-ms-transform:translate3d(0,50px,0);transform:translate3d(0,50px,0);opacity:0;visibility:hidden;}"
    styleId="2628240846"
  />
</div>
`;
